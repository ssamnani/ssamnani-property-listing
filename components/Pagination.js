import "../styles/Pagination.module.css";

function Pagination(props) {
  const { pageStart, pageEnd, totalRecords } = props;

  return (
    <div className="paginationWrapper">
      <div className="paginationInfo">
        Viewing {pageStart}-{pageEnd} of {totalRecords} property results
      </div>
      <div className="paginationButtons">
        <ul>
          <li className="active">1</li>
          <li>2</li>
          <li>3</li>
          <li>4</li>
          <li>&gt;</li>
        </ul>
      </div>
    </div>
  );
}

export default Pagination;
