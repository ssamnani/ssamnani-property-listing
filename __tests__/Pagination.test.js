import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Pagination from "../components/Pagination";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render("<Pagination />", div);
});

// Snapshot Testing
it("matches snapshot", () => {
  const tree = renderer
    .create(<Pagination pageStart="1" pageEnd="20" totalRecords="43543" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
