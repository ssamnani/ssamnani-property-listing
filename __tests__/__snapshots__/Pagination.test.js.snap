// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="paginationWrapper"
>
  <div
    className="paginationInfo"
  >
    Viewing 
    1
    -
    20
     of 
    43543
     property results
  </div>
  <div
    className="paginationButtons"
  >
    <ul>
      <li
        className="active"
      >
        1
      </li>
      <li>
        2
      </li>
      <li>
        3
      </li>
      <li>
        4
      </li>
      <li>
        &gt;
      </li>
    </ul>
  </div>
</div>
`;
